src/cli/rustup_mode.rs

```diff
@@ -1312,10 +1318,28 @@ fn toolchain_link(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
 }
 
 fn toolchain_remove(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
-    for toolchain in m.values_of("toolchain").unwrap() {
-        let toolchain = cfg.get_toolchain(toolchain, false)?;
-        toolchain.remove()?;
+    if m.is_present("regex") {
+        assert!(
+            m.values_of("toolchain").unwrap().len() == 1,
+            "exactly one regex filter must be supplied"
+        );
+
+        let regex = regex::Regex::from_str(m.values_of("toolchain").unwrap().next().unwrap())
+            .expect("invalid regex");
```

Comment:
Same here.

Reply:
If it were a simple pattern rather than a regex this wouldn't be necessary.  Otherwise it ought to be a clap validator function checking this.

Reply:
Actually you can have invalid patterns. I added an error case for it.